back/app/services/user_custom_field_service.rb

```diff
@@ -1,10 +1,26 @@
 # frozen_string_literal: true
 
 class UserCustomFieldService
-  def delete_custom_field_values(field)
-    User
-      .where("custom_field_values \? '#{field.key}'")
-      .update_all("custom_field_values = custom_field_values - '#{field.key}'")
+  def delete_custom_field_values(field, current_user)
+    users = User.where("custom_field_values ? '#{field.key}'")
+    log = {}
+
+    users.each do |user|
+      log[user.id] = { field.key.to_s => user.custom_field_values[field.key] }
+    end
```

Comment:
Is this a significant performance hit (with large n of users selected?). If 'yes', would plucking data and passing to job after we update the users be significant improvement? Need to test this.

Reply:
Nope, almost no difference, and with either method we're looking at about 1/100th second to process 1k users, so tolerable.